### ISSUE TITLE
Rucio client seems to be several steps behind what FTS reports.

### ISSUE THREAD

**Issue by jhonatanamado** [dmwm/CMSRucio#251]
This is related to the transfer test and is not the first time that we see this issue in transfers from Tier0.
Tier0 spotted a  huge time mismatch between what FTS report to the monitoring and what Rucio sees.  Take for example [this](https://monit-grafana.cern.ch/d/_k0w6fcGz/cms-tier0-replay-vocms015?orgId=11&from=1645560000000&to=1645714800000&viewPanel=18) plot called transfer backlog by site which is using the `rucio client replica`  to check if a replica is `Available` or different from `Available`  from transfers related to one of our accounts

<img width="1597" alt="Backlog" src="https://user-images.githubusercontent.com/50465903/155752812-94617ccb-16df-4f09-8811-326a98b54baa.png">

And now lets check what [FTS](https://monit-grafana.cern.ch/d/mtQFDScGk/cms-fts-metrics?orgId=11&var-vo=cms&var-group_by=activity&var-src_rse=T0_CH_CERN_Disk&var-dst_rse=T1_US_FNAL_Tape&var-fts_server=All&var-activity=All&var-protocol=All&var-bin=1h&from=1645560000000&to=1645714800000) report, in the same time window.

<img width="1588" alt="TransferRate" src="https://user-images.githubusercontent.com/50465903/155752981-ddd26a32-cd69-4408-a057-a3de2953ca7e.png">

Indeed FNAL_Tape shows a good throughput and doing a quick math taking the average of that throughput when the transfers were ongoing is around the same as the max backlog for FNAL_Tape.  But check the panel called transfer backlog, it seems that Rucio is reporting slow transfers to FNAL_Tape.
The problem is that the panel called transfer backlog should show a steeper curve. It seems that rucio takes time to distinguish if the replica has already arrive to the destination. As if Rucio was several steps behind what FTS reports.

Data deletion in tier0 is based on what Rucio Client reports, if rucio is several steps behind what FTS reports during data taking, we might not be deleting data fast enough and will be a huge issue for Tier0. 

Checking the number of pods running shows the following.

```
daemons-conveyor-finisher-69c69d6648-9j6gr                   1/1       Running   0          3d
daemons-conveyor-finisher-69c69d6648-bv6v6                   1/1       Running   0          3d
daemons-conveyor-finisher-69c69d6648-mvhgj                   1/1       Running   0          3d
daemons-conveyor-finisher-69c69d6648-qn7qk                   1/1       Running   0          3d
daemons-conveyor-poller-848df97775-67jh9                     1/1       Running   0          3d
daemons-conveyor-poller-848df97775-ttjr2                     1/1       Running   0          3d
daemons-conveyor-submitter-656b458d4b-j2x67                  1/1       Running   0          3d
daemons-conveyor-submitter-656b458d4b-vnn9z                  1/1       Running   0          3d
daemons-conveyor-submitter-656b458d4b-w2hql                  1/1       Running   1          3d
loadtest-daemons-conveyor-submitter-866569cc44-2pt2m         1/1       Running   0          3d
```
Felipe Gomez @flgomezc from DataManagement have seen this behaviour for other transfers but possibly they did not have the evidence necessary to create this issue.

**Comment by ericvaandering** [comment]
I'd start by looking at either the logs or better if there is relative monitoring coming out of the poller and the finisher. I think the sloping plot you show is not "complete" until the finisher has completed work on some file. There are probably three knobs we can turn for each of these, number of pods, number of threads, and hopefully a chunk size. Increasing the chunk size would be easiest on the database and might help us complete things faster. That's the kind of issue we had with the reaper too.

**Comment by yuyiguo** [comment]
How often do conveyor-poller  retrieve the transfer state information from FTS? Is this configurable?

**Comment by jhonatanamado** [comment]
Ok, So I believed the transfers are not done until the `archived` bit is set. (is this bit also set for buffer?) If that the case, all the FTS monitoring for tape sites shows values to only buffer? Am I correct?

**Comment by DAMason** [comment]
That has been my interpretation.  I thought the archived bit is set once the additional check that the file has been physically written to tape happens.  I don't know to what extent that is site dependent, but could imagine it to be so.

**Comment by ericvaandering** [comment]
Either we have set that check for every site or it is set for all tape transfers. Regardless, no where in CMS are FTS transfers truly done until the data is safely on tape. Also, Dave I think I've restored the monitoring you were interested in. So I'm going to close this issue as a red herring. Dave, thanks for remembering the archiving to tape part. 

Any further problems, of course, reopen the issue.

**Comment by nsmith-** [comment]
The FTS state transitions lower down in that page give some indication that the bandwidth plot is only counting transfers that are active. After the file reaches the tape buffer, the FTS job moves into the `ARCHIVING` state, in which it lasts for some time longer before finishing (I don't know why the `FINISHED` transition is so low, perhaps because the FTS job is deleted/removed shortly after)
<img width="945" alt="Screen Shot 2022-02-28 at 1 23 42 PM" src="https://user-images.githubusercontent.com/6587412/156045587-bb6d7207-b86a-49aa-8ee1-69ebc9418871.png">
It looks like the number of `ARCHIVING` jobs falls off around the same time as the rucio queue is emptied, so I think this is consistent with Dave's interpretation.

TL;DR rucio is right
